remove scripts, implement correctly cairosvg

bil-elmoussaoui opened the pull request:
Things i've done till now : 
- Remove both scripts
- Use Inkscape if Cairosvg is not installed
- Check if `png2icns` is installed for OSX conversion

Things still not working correctly
- `svg2png` does not convert svg files correctly
- still can't scale svg icons using `svg2png`

I will open an other PR to fix those

To test the script using `inkscape` (everything works fine with it)  add `use_inkscape = True` after the line 24

Foggalong commented:
I can't merge this into the master branch if it's not fully working yet. I've made a [python branch](https://github.com/numixproject/core-demo/tree/python) which you can merge it into though :)

bil-elmoussaoui commented:
Yeah i just pushed the modification in case i lose all my files :p I will push the modifications to this commit

bil-elmoussaoui commented:
I did some search and i found out that the opacity issue is due to cairosvg issue that has not been fixed.. (latest version was in 2012). I found an equivalent `pyrsvg` but this library is only working with python2 :/ 
We have two solutions either we look for an other library or a way to use Cairo directly (which will be hard..) or find a better library, what do you think?

Foggalong commented:
Do any other libraries exist?

bil-elmoussaoui commented:
The only library that we can use is Wand and it uses ImageMagick to convert svg's to png files, it works but the conversion quality is very poor.. see the comparaison

**Scaled to 196x196 and converted using Wand**
![com danvelazco fbwrapper](https://cloud.githubusercontent.com/assets/7660997/15271339/e1433e16-1a43-11e6-8a6d-36749785bf25.png)
**Scaled to 196x196 and converted to png using Inkscape**
![com danvelazco fbwrapper](https://cloud.githubusercontent.com/assets/7660997/15271342/065c1722-1a44-11e6-8426-8fec825aece9.png)

bil-elmoussaoui commented:
Fixed!
New result : 
![com danvelazco fbwrapper](https://cloud.githubusercontent.com/assets/7660997/15271699/e42f2b18-1a56-11e6-97c9-af89f63a308a.png)
I used Cairo and Rsvg from `gi.repository`

Foggalong commented:
That's brilliant! Does this fix all the issues stated above then?

bil-elmoussaoui commented:
Yes, everything works perfectly!

Foggalong commented on gen.py:
```diff
@@ -38,9 +44,40 @@ def mkdir(dir):
-    """Context manager for changing the current working directory"""
+
     def __init__(self, newPath):
+        """Context manager for changing the current working directory"""
```
Curious: why this comment move?

bil-elmoussaoui replied:
Not me, i just used an atom extension that "beautify" the source code, and it moves the class comment to it's constructor so it can be shown if you an IDE

Foggalong replied:
Ah okay, that makes sense :)

Foggalong commented on gen.py:
```diff
@@ -62,39 +99,40 @@ def __exit__(self, etype, value, traceback):
+            root = icons[icon]["linux"]["root"] + ".svg"
+            copy2("icons/" + size + "/" + icon + ".svg",
+                  ldir + size + "/apps/" + root)
+        with cd(ldir + size + "/apps/"):
```
Is this the only place where `cd` is used now?

bil-elmoussaoui replied:
Yeah, i can replace it with a normal for to reduce the code length if you want that :)

Foggalong replied:
Yeah, definitely if it's only being used in one place now.

bil-elmoussaoui replied:
Done
